streams/src/test/java/org/apache/kafka/streams/processor/internals/StreamsProducerTest.java

```diff
@@ -351,22 +404,12 @@ public void shouldSkipAbortTxOnEosAbortIfNotTxInFlight() {
 
     // error handling tests
 
-    @Test
-    public void shouldFailIfApplicationIdIsNullOnEos() {
```

Comment:
This test does not apply any longer, as we don't pass in `application.id` into the constructor any more.